test/e2e/azure_machinepool_drain.go

```diff
@@ -157,13 +166,39 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 			return err
 		}
 
-		decreasedReplicas := *owningMachinePool.Spec.Replicas - int32(1)
+		decreasedReplicas = *owningMachinePool.Spec.Replicas - int32(1)
 		owningMachinePool.Spec.Replicas = &decreasedReplicas
 		return helper.Patch(ctx, owningMachinePool)
 	}, 3*time.Minute, 3*time.Second).Should(Succeed())
 
-	// TODO setup a watcher to validate expected 2nd order drain outcomes
-	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159
+	By(fmt.Sprintf("checking for a machine to start draining for machine pool: %s/%s", amp.Namespace, amp.Name))
+	Eventually(func() error {
+		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
+		if err != nil {
+			LogWarning(errors.Wrap(err, "failed to list the azure machine pool machines").Error())
+			return err
+		}
+
+		for i := range ampmls {
+			if conditions.IsFalse(&ampmls[i], clusterv1.DrainingSucceededCondition) {
```

Comment:
why are we checking if the condition is False here?

Reply:
I adapted this line from the previous iteration of the cordon and drain test, which checked if the condition is false.. However it does make a lot more sense to check if it is true. The blame here mentions the original version of this test checks to see if the condition is false: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blame/9629164862fc682181e579431d28a84186d6ceaa/test/e2e/azure_machinepool_drain.go#L179

However, in the original commit, the diff shows the condition being checked if it is true... I'm wondering what happened here and whether there was a reason it was changed to `isFalse` 🤔 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/d1c01511604f40360cfa7db9ffc61dabec3e702a#diff-ef1035bcafba9959d0df4cb4d9544deafccfb9d9a9f7e89cbcf1cb8565a89880R189

Reply:
None of the history for this file showed that the `isTrue` was changed to `isFalse` prior to the code being removed. Is this even possible? https://github.com/jackfrancis/cluster-api-provider-azure/commits/c2781f00104ad84b22934c72cf56e22c50beabf0/test/e2e/azure_machinepool_drain.go

For now, I'll go ahead and set it to `isTrue` as it would make more sense to check if the condition is true.